include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -0,0 +1,111 @@
+    const uint32_t hi_y_ix = hi_x_ix + 2 * WorkgroupSize;
+
+    sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+    complex_t<Scalar> lo = {sharedmemAccessor.get(lo_x_ix), sharedmemAccessor.get(lo_y_ix)};
+    complex_t<Scalar> hi = {sharedmemAccessor.get(hi_x_ix), sharedmemAccessor.get(hi_y_ix)};
+
+    fft::DIX<Scalar, inverse>::radix2(fft::twiddle<Scalar, inverse>(sub_ix, step << 1), lo, hi);
+
+    // Share results between threads
+    sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+    sharedmemAccessor.set(lo_x_ix, lo.real());
+    sharedmemAccessor.set(hi_x_ix, hi.real());
+    sharedmemAccessor.set(lo_y_ix, lo.imag());
+    sharedmemAccessor.set(hi_y_ix, hi.imag());     
+}
+
+// When doing a 1D FFT of size N, we call N/2 threads to do the work, where each thread of index 0 <= k < N/2 is in charge of computing two elements
+// in the output array, those of index k and k + N/2. The first is "lo" and the latter is "hi"
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize, uint32_t SubgroupSize, bool inverse>
+void FFT(NBL_REF_ARG(Accessor) sharedmemAccessor, NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi){
+    const uint32_t doubleWorkgroupSize = WorkgroupSize << 1;
+    const uint32_t threadID = glsl::gl_LocalInvocationIndex();
+
+    if (inverse)
+    {
+        // Run a subgroup-sized FFT, then continue with bigger steps
+        subgroup::FFT<Scalar, SubgroupSize, true>(lo, hi);
+        
+        // Put values into workgroup shared smem
+        sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+        store<Accessor, Scalar, WorkgroupSize>(sharedmemAccessor, threadID, lo, hi);             
+
+        // The bigger steps
+        for (uint32_t step = SubgroupSize << 1; step < WorkgroupSize; step <<= 1)
+            FFT_loop<Accessor, Scalar, WorkgroupSize, true>(sharedmemAccessor, step, threadID);
+
+        // Get values back from smem
+        sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+        load<Accessor, Scalar, WorkgroupSize>(sharedmemAccessor, threadID, lo, hi); 
+        sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+
+        // special last iteration
+        fft::DIT<Scalar>::radix2(fft::twiddle<Scalar, true>(threadID, doubleWorkgroupSize), lo, hi); 
+        divides_assign< complex_t<Scalar> > divAss;
+        divAss(lo, WorkgroupSize / SubgroupSize);
+        divAss(hi, WorkgroupSize / SubgroupSize);
+    }
+    else
+    {
+        // special first iteration
+        fft::DIF<Scalar>::radix2(fft::twiddle<Scalar, false>(threadID, doubleWorkgroupSize), lo, hi); 
+
+        // Put values into workgroup shared smem
+        sharedmemAccessor.workgroupExecutionAndMemoryBarrier();
+        store<Accessor, Scalar, WorkgroupSize>(sharedmemAccessor, threadID, lo, hi);
+        
+        // Run bigger steps until Subgroup-sized
+        for (uint32_t step = WorkgroupSize >> 1; step > SubgroupSize; step >>= 1)
```

Comment:
how does it make sense to start at `step=WorkgroupSize*2` ?

That step is makes you step over the whole  array